### DISCUSSION TITLE
Consumers with the same key

### DISCUSSION THREAD

**Discussion by poostwoud** [apache/apisix#10810] [Q&A]
Hi,

How does apisix distinguish between consumers having the same key defined when using the key-auth plugin?

Cheers,

Peter

**Comment by kayx23** [comment]
Hi @poostwoud , it is differentiated through `username`.

**Reply by poostwoud** [reply]
Hi @kayx23, but there is no username involved in the key-auth plugin right? Looking at https://github.com/apache/apisix/blob/2d47b4b3b33730b1de92c91c4fbe2b239ee339af/apisix/plugins/key-auth.lua#L88 it seems the consumer is identified only by the key?

**Reply by kayx23** [reply]
What I meant was the plugin details would need to be configured on a consumer and consumers are identified by username.

**Reply by ch-gr** [reply]
@kayx23 I'm not sure what that means.

If you access a route, supplying a key-auth header, but there are two consumers that have key-auth, both using the same key you supplied, what is the expected outcome?

Looks like that is discussed here, there's no solution yet:
- https://github.com/apache/apisix/issues/11197
